internal/pkg/aws/codepipeline/codepipeline.go

```diff
@@ -114,12 +117,15 @@ func New(s *session.Session) *CodePipeline {
 
 // GetPipeline retrieves information from a given pipeline.
 func (c *CodePipeline) GetPipeline(name string) (*Pipeline, error) {
-	input := &cp.GetPipelineInput{
+	resp, err := c.client.GetPipeline(&cp.GetPipelineInput{
 		Name: aws.String(name),
-	}
-	resp, err := c.client.GetPipeline(input)
-
+	})
 	if err != nil {
+		var notFoundErr *cp.PipelineNotFoundException
+		if errors.As(err, &notFoundErr) {
+			return nil, fmt.Errorf("get pipeline %s: %w", name, ErrPipelineNotFound)
```

Comment:
nit: the clients will most likely wrap w `get pipeline` then this second one becomes a bit redundant
```suggestion
			return nil, ErrPipelineNotFound
```

Reply:
Can we add a test case to capture this error

Reply:
jk. i actually removed this error since i'm using `ListPipelineNamesByTags()` instead now